code/game/objects/items/RCD.dm

```diff
@@ -276,6 +402,123 @@ GLOBAL_VAR_INIT(icon_holographic_window, init_holographic_window())
 	. = ..()
 	AddElement(/datum/element/openspace_item_click_handler)
 
+	//Create the radial menu as a static list only at the beginning of the map for all RCD instances to share
+	if(!mapload)
+		return
+
+	var/list/icon_list = list()
+	var/list/essentials = list(
```

Comment:
All of this:

1. Needs to be in its own proc
2. Needs to be done only once
3. Should only be done when necessary.

This is all extremely heavy to compute.

Reply:
Why create a proc and have extra method calling overhead when that proc would only be called by Initialize() anyway?. Also this stuff is only invoked once. See the if statement above

```
//Create the radial menu as a static list only at the beginning of the map for all RCD instances to share
	if(!mapload)
		return
```

mapload will be false for all instances of RCD's created after the map has loaded and i store it in a static list so memory isnt a issue either